### PR TITLE
DEV-2898 | trigger thank you email receipts via webhooks instead of through interstitial view (backend changes for DEV-2744)

### PR DESCRIPTION
**Pull request by benjaminEwhite**
Please fill out each section even if it's just with "N/A"

#### Plan? (if this draft/incomplete indicate what you intend to do and how)

This branch and PR for for one of the two subtasks/tickets that will together accomplish the AC for DEV-2744.  Both [DEV-2898](https://news-revenue-hub.atlassian.net/browse/DEV-2898) (which will contain the front end changes required for 2744), and this PR will be merged into DEV-2744, which will itself then be merged into main.

QA will happen only for 2744, and not for the subtask PRs.

Teardown for PRs for 2898 and 2899 should happen immediately after merging each into DEV-2744.

#### What's this PR do?

- Removes the `payment_success` view and associated tests
- Refactors webhook receiver that handles payment intent success to call `contribution.handle_thank_you_email` and updates tests accordingly.
- Also, this branch merged in changes from DEV-3239 which fixes bug with creating Hookdeck connections for Stripe webhooks in review app. Those changes were necessary to get the review app to work so I could ensure that the feature worked as expected (even though QA will happen on parent ticket). 3239 will be merged into main before this (and 2744) so that diff will just go away. I will note the files that can be ignored in "files changed" view.

#### Why are we doing this? How does it help us?

Avoid double sending thank you emails.

#### How should this be manually tested? Please include detailed step-by-step instructions.

Try making both approved and flagged + approved one-time and recurring contributions on a page whose org has enabled thank you emails.

You should get a *single* thank you email. 

Furthermore, if you hit the "back" button after getting taken to the donation page, you should get taken to the contribution page straight away, and not to the interstitial payment success view in the SPA like you don in the status quo.

#### Did this PR make changes to the user interface that may require changes to the user-facing docs?

No

#### Have automated unit tests been added? If not, why?

Yes

#### How should this change be communicated to end users?

N/A

#### Are there any smells or added technical debt to note?

No

#### Has this been documented? If so, where?

N/A

#### What are the relevant tickets? Add a link to any relevant ones.

- [DEV-2898](https://news-revenue-hub.atlassian.net/browse/DEV-2898) 
- [DEV-2899](https://news-revenue-hub.atlassian.net/browse/DEV-2899) 
- [DEV-2744](https://news-revenue-hub.atlassian.net/browse/DEV-2744) 

#### Do any changes need to be made before deployment to production (adding environment variables, for example)?

No

#### Are there next steps? If so, what? Have tickets been opened for them? List next-step tickets here:

Already described in "Plan" above re: stacked PR strategy.

[DEV-2898]: https://news-revenue-hub.atlassian.net/browse/DEV-2898?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DEV-2898]: https://news-revenue-hub.atlassian.net/browse/DEV-2898?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by benjaminEwhite** [comment]
Closing this PR, tearing down review app, then re-creating. This is because webhooks were not correctly configured when bootstrap review app command ran.

File: apps/users/management/commands/bootstrap-review-app.py [hunk 1]
```diff
@@ -16,15 +15,13 @@ class Command(BaseCommand):  # pragma: no cover low ROI for test of command line
...
 
     def handle(self, *args, **options):
-        branch_name = os.getenv("HEROKU_BRANCH")
-        zone_name = os.getenv("CF_ZONE_NAME")
-        heroku_app_name = os.getenv("HEROKU_APP_NAME")
-        heroku_api_key = os.getenv("HEROKU_API_KEY")
+        branch_name = settings.HEROKU_BRANCH
+        zone_name = settings.CF_ZONE_NAME
+        heroku_app_name = settings.HEROKU_APP_NAME
+        heroku_api_key = settings.HEROKU_API_KEY
```
Comment (benjaminEwhite) [inline apps/users/management/commands/bootstrap-review-app.py:21]: These changes are part of [DEV-3239](https://news-revenue-hub.atlassian.net/browse/DEV-3239) which solves for bug with review app hookdeck configuration. Needed to pull them in here to get review app working to confirm feature works when deployed. This diff will go away before this PR is merged, and doesn't require review here.

File: apps/users/tests/test_management.py [hunk 2]
```diff
@@ -1,50 +1,62 @@
...
 from apps.organizations.tests.factories import RevenueProgramFactory
 
 
-@pytest.mark.django_db
-def test_bootstrap_review_app_bootstraps_hookdeck(monkeypatch):
-    heroku_branch_name = "DEV-9999--foo-barrr"
-    heroku_app_name = "rev-engine-dev-9999-some-uid"
-    monkeypatch.setenv("HEROKU_BRANCH", heroku_branch_name)
-    monkeypatch.setenv("HEROKU_APP_NAME", heroku_app_name)
+MOCK_TICKET_ID = "DEV-9999"
```
Comment (benjaminEwhite) [inline apps/users/tests/test_management.py:12]: The changes in this file are part of [DEV-3239](https://news-revenue-hub.atlassian.net/browse/DEV-3239) which solves for bug with review app hookdeck configuration. Needed to pull them in here to get review app working to confirm feature works when deployed. This diff will go away before this PR is merged, and doesn't require review here.

File: revengine/settings/base.py [hunk 3]
```diff
@@ -434,6 +434,7 @@
 
 ### Heroku Settings
 HEROKU_APP_NAME = os.getenv("HEROKU_APP_NAME")
+HEROKU_API_KEY = os.getenv("HEROKU_API_KEY")
```
Comment (benjaminEwhite) [inline revengine/settings/base.py:437]: The changes in this file are part of [DEV-3239](https://news-revenue-hub.atlassian.net/browse/DEV-3239) which solves for bug with review app hookdeck configuration. Needed to pull them in here to get review app working to confirm feature works when deployed. This diff will go away before this PR is merged, and doesn't require review here.

**Review by njharman — APPROVED** [review]
Code looks good. 

In "manual test" section of PR I suggest making it more clear that you should receive a **single** email / verify only one email sent.

**Pull request merged**